src/components/admin/MenuBar.vue

```diff
@@ -1,19 +1,48 @@
 <script setup lang="ts">
+  import { useAuthStore, type AuthStore } from '@/stores/AuthStore';
+  import {
+    usePersonenkontextStore,
+    type PersonenkontextStore,
+    type SystemrechtResponse,
+  } from '@/stores/PersonenKontextStore';
   import { type Ref, ref, type ComputedRef } from 'vue';
   import { onMounted } from 'vue';
   import { useDisplay } from 'vuetify';
 
   const menuDrawer: Ref<boolean> = ref(true);
   const { mobile }: { mobile: ComputedRef<boolean> } = useDisplay();
 
+  const auth: AuthStore = useAuthStore();
+  const personenkontextStore: PersonenkontextStore = usePersonenkontextStore();
+
+  const hasRollenverwaltungRecht: Ref<boolean> = ref(false);
+
   function closeMenuOnMobile(): void {
     if (mobile.value) {
       menuDrawer.value = false;
     }
   }
 
-  onMounted(() => {
-    menuDrawer.value = !mobile.value;
+  // Retrieves logged in user info on mounted and check using the user's sub if he has the right "ROLLEN_VERWALTEN". If not (error from the endpoint) then
+  // hide the menu from the menubar.
+  onMounted(async () => {
+    await auth.getLoggedInUserInfo();
+    try {
+      if (auth.currentUser && auth.currentUser.personId) {
+        const response: SystemrechtResponse = await personenkontextStore.hasSystemrecht(
```

Comment:
Ich stelle immer noch in Frage ob es wirklich die richtige Lösung ist, die Systemrechte einzeln abzufragen. @kristoff-kiefer, weißt du wie viele Systemrechte da noch kommen? Wenn für jedes Recht eine eigene Abfrage abgeschickt werden muss, ist das ungünstig.

Reply:
Denke Ich auch :/ wäre meiner Sicht viel besser alle Rechte auf einmal abzufragen. Was spricht eigentlich dagegen?

Reply:
Ich versuche das in einem anderen Scope mit Kristoff zu besprechen, da der Endpunkt überarbeitet werden müsste und das auch unabhängig von diesem Scope passieren kann.

Reply:
Für die Abfrage im Frontend isses das eigentlich schon. Meistens (eigentlich immer) werden sie im Backend schon zurechtgeschoben